src/OrchardCore/OrchardCore.DisplayManagement/TagHelpers/AddPropertyTagHelper.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using System.Threading.Tasks;
+
+namespace OrchardCore.DisplayManagement.TagHelpers;
+
+[HtmlTargetElement("add-property", Attributes = "name", TagStructure = TagStructure.NormalOrSelfClosing)]
+public class AddPropertyTagHelper : TagHelper
+{
+    [HtmlAttributeName("name")]
+    public string Name { get; set; }
+
+    public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+    {
+        var shape = (IShape)context.Items[typeof(IShape)];
+
+        if (!string.IsNullOrWhiteSpace(Name))
+        {
+            var content = await output.GetChildContentAsync(useCachedResult: false);
+            shape.Properties[Name.Trim()] = output.Attributes.ContainsName("value")
+                ? output.Attributes["value"].Value
+                : new HtmlString(content.GetContent());
+        }
+
+        output.SuppressOutput();
```

Comment:
```suggestion
        if (string.IsNullOrWhiteSpace(Name))
        {
            output.SuppressOutput();
        }

        var content = await output.GetChildContentAsync(useCachedResult: false);
        var shape = (IShape)context.Items[typeof(IShape)];
        shape.Properties[Name.Trim()] = output.Attributes.ContainsName("value")
            ? output.Attributes["value"].Value
            : new HtmlString(content.GetContent());
```

Reply:
Could you explain this suggestion? It's not equivalent, because the `if` doesn't return so now the content is _always_ called. Did you mean this?
```suggestion
        if (string.IsNullOrWhiteSpace(Name))
        {
            output.SuppressOutput();
            return;
        }

        var content = await output.GetChildContentAsync(useCachedResult: false);
        var shape = (IShape)context.Items[typeof(IShape)];
        shape.Properties[Name.Trim()] = output.Attributes.ContainsName("value")
            ? output.Attributes["value"].Value
            : new HtmlString(content.GetContent());
            
        output.SuppressOutput();
```

Reply:
I updated the suggestion